velox/functions/prestosql/ArithmeticImpl.h

```diff
@@ -44,10 +44,12 @@ round(const TNum& number, const TDecimals& decimals = 0) {
   }
 
   double factor = std::pow(10, decimals);
+  static const TNum kInf = std::numeric_limits<TNum>::infinity();
   if (number < 0) {
-    return (std::round(number * factor * -1) / factor) * -1;
+    return (std::round(std::nextafter(number, -kInf) * factor * -1) / factor) *
+        -1;
   }
-  return std::round(number * factor) / factor;
+  return std::round(std::nextafter(number, kInf) * factor) / factor;
```

Comment:
It does not look quite right to me, can you double check how it is done in Presto Java?

Reply:
Thanks. will check.

Reply:
I found the implementation.  [code link](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java#L1132-L1147)
presto use java Math.round function. And I write a small example.
```
public class Main{
    public static void main(String []args) {
       double num = 0.57499999999999996 * 100;
       double roundNum = Math.round(num);
       System.out.println(roundNum);
       System.out.println(roundNum/100);
    }
} 
```
Output:
```
57.0
0.57
```

Reply:
The essence of this problem is that floating point numbers are not precise, we should use decimal, do you think it is ok? @Yuhta 
Here is spark code implementation of double/float round. https://github.com/apache/spark/blob/branch-3.2/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala#L1356-L1367

Reply:
spark and presto have same semantic for round function. 
Spark sql:
![image](https://github.com/facebookincubator/velox/assets/14065538/d77497cc-4d7e-4672-b2be-c939aac8a6d0)
Presto sql:
![image](https://github.com/facebookincubator/velox/assets/14065538/19eac98a-3ca4-4b3d-85b5-d6a3597bbfc7)

Reply:
I think they fixed it in SQL parser, the old code of `round` looks correct to me.
Changing to decimal would be too expensive here.

Reply:
You are right. I will create a new PR to add a round function for spark.

Reply:
I create https://github.com/facebookincubator/velox/pull/5992 to add a new round function for spark.